src/drivers/intel/hda/hda-dma.c

```diff
@@ -546,10 +538,15 @@ static int hda_dma_start(struct dma_chan_data *channel)
 		goto out;
 	}
 
-	ret = hda_dma_enable_unlock(channel);
+	irq_local_enable(flags);
+
+	ret = hda_dma_enable(channel);
```

Comment:
to understand why `hda_dma_enable()` can be safely called with interrupts enabled, whereas interrupts must be disabled when reading DGCS above, I'd need to understand the controller well. At least we need a comment here, but naively this doesn't look right.